core/src/main/scala/kafka/server/ReplicaManager.scala

```diff
@@ -1144,10 +1147,10 @@ class ReplicaManager(val config: KafkaConfig,
   def becomeLeaderOrFollower(correlationId: Int,
                              leaderAndIsrRequest: LeaderAndIsrRequest,
                              onLeadershipChange: (Iterable[Partition], Iterable[Partition]) => Unit): LeaderAndIsrResponse = {
-    leaderAndIsrRequest.partitionStates.asScala.foreach { case (topicPartition, stateInfo) =>
-      stateChangeLogger.trace(s"Received LeaderAndIsr request $stateInfo " +
+    leaderAndIsrRequest.partitionStates.asScala.foreach { partitionState =>
```

Comment:
We should wrap this in an `if (stateChangeLogger.isTraceEnabled)` so that we don't execute the "for" loop unless there is actually something to do in the loop body.  This is an existing issue, of course, and probably better as a follow-on change.